### PR TITLE
[IMP] mail: activity menu no activity

### PR DESCRIPTION
**Pull request by tsm-odoo**

### CODE REVIEW

File: addons/mail/static/src/activity/activity_menu.scss [hunk 1]
```diff
@@ -11,3 +13,12 @@
         background-color: map-get($theme-colors, "light");
     }
 }
+
+.o-mail-no-activity {
+    cursor: initial;
```
Comment (alexkuhn) [inline addons/mail/static/src/activity/activity_menu.scss]: Don't see any reason for this `cursor: initial` style rule? If unused, then drop.

File: addons/mail/static/src/activity/activity_menu.scss [hunk 2]
```diff
@@ -11,3 +13,12 @@
...
+    align-items: center;
```
Comment (alexkuhn) [inline addons/mail/static/src/activity/activity_menu.scss]: You can use `align-items-center` classname.

File: addons/mail/static/src/activity/activity_menu.scss [hunk 3]
```diff
@@ -11,3 +13,12 @@
...
+    color: grey;
+    opacity: 0.5;
+    padding: 3px;
```
Comment (alexkuhn) [inline addons/mail/static/src/activity/activity_menu.scss]: Could use `p-2` classname.

File: addons/mail/static/src/activity/activity_menu.scss [hunk 4]
```diff
@@ -11,3 +13,12 @@
         background-color: map-get($theme-colors, "light");
     }
 }
+
+.o-mail-no-activity {
+    cursor: initial;
+    align-items: center;
+    color: grey;
```
Comment (alexkuhn) [inline addons/mail/static/src/activity/activity_menu.scss]: Prefer a `text-muted` classname, which is more the intention and is dark-theme compliant when enterprise branch will work eventually.

File: addons/mail/static/src/activity/activity_menu.scss [hunk 5]
```diff
@@ -1,6 +1,8 @@
 .o-mail-activity-menu {
     width: 350px;
     max-height: 400px;
+    min-height: 50px;
+    flex: 0 1 auto;
```
Comment (alexkuhn) [inline addons/mail/static/src/activity/activity_menu.scss]: Why precise `flex: 0 1 auto`?
That's the initial value, and I don't see any other style rule affecting these styling properties.

File: addons/mail/static/src/activity/activity_menu.scss [hunk 6]
```diff
@@ -11,3 +13,12 @@
         background-color: map-get($theme-colors, "light");
     }
 }
+
+.o-mail-no-activity {
+    cursor: initial;
+    align-items: center;
+    color: grey;
+    opacity: 0.5;
```
Comment (alexkuhn) [inline addons/mail/static/src/activity/activity_menu.scss]: Can use `opacity-50` classname.

File: addons/mail/static/src/activity/activity_menu.xml [hunk 7]
```diff
@@ -8,29 +8,34 @@
...
-                                <span t-else="" class="text-muted py-0 px-2">0 Late</span>
-                                <button t-if="group.today_count" type="button" class="btn btn-link py-0 px-2" t-on-click.stop="() => this.openActivityGroup(group, 'today')"><t t-esc="group.today_count"/> Today</button>
-                                <span t-else="" class="text-muted py-0 px-2">0 Today</span>
-                                <span class="flex-grow-1"/>
-                                <button t-if="group.planned_count" type="button" class="btn btn-link py-0 px-2" t-on-click.stop="() => this.openActivityGroup(group, 'upcoming_all')"><t t-esc="group.planned_count"/> Future</button>
-                                <span t-else="" class="text-muted py-0 px-2">0 Future</span>
+            <div class="o-mail-activity-menu">
+                <div t-if="activity.counter === 0" class="o-mail-no-activity dropdown-item-text text-center d-flex justify-content-center">
```
Comment (alexkuhn) [inline addons/mail/static/src/activity/activity_menu.xml]: `text-center` is useless with `d-flex align-items-center`, as you use flexbox alignment instead.

**Review by alexkuhn — COMMENTED** [review]
LGTM otherwise

**Comment by alexkuhn** [comment]
Would be also nice to make a test.
It's not super useful here but it'd be good to take the habit of making one whenever possible.

@ged-odoo I suppose we should introduce an `ActivityServer` with specific `snakeCaseRoute` like `_web_dataset_call_kw_res_users_systray_get_activities`?
As activity menu is coupled to `mail.activity` rather than `mail.messaging`, would make sense to keep same mapping of service/server.

**Comment by ged-odoo** [comment]
@alexkuhn not sure, some routes such as `/mail/thread/data` kind of mix activity with other stuff. I guess it would be easier, for now, to just keep 1 test server. maybe we can rename it to TestServer, to avoid confusion?

File: addons/mail/static/src/activity/activity_menu.scss (hunk 5 above)
Comment (tsm-odoo) [reply addons/mail/static/src/activity/activity_menu.scss]: that's a mistake :)

**Comment by tsm-odoo** [comment]
Hello @alexkuhn, I applied your suggestions. Can those changes be merged now ?

File: addons/mail/static/tests/activity/activity_menu_tests.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+    QUnit.module("activity menu");
+
+    QUnit.test("activity menu: no activity", async (assert) => {
+        const server = new MessagingServer();
+        const env = makeMessagingEnv((route, params) => server.rpc(route, params));
+        await mount(ActivityMenu, target, { env });
+        await click(document.querySelector("i[aria-label='Activities']").closest('.dropdown-toggle'));
+        assert.containsOnce(target, ".o-mail-no-activity");
+        assert.strictEqual(target.querySelector(".o-mail-no-activity span").innerText, "Congratulations, you're done with your activities.");
```
Comment (alexkuhn) [inline addons/mail/static/tests/activity/activity_menu_tests.js]: Would not assert explicit text (the classname `o-mail-no-activity` is good enough)

**Review by alexkuhn — COMMENTED** [review]
LGTM otherwise

**Pull request merged**